### DISCUSSION TITLE
Unable to Set Persistent Name for HomeKit Device

### DISCUSSION THREAD

**Discussion by kdorepos** [homebridge-simplisafe3/homebridge-simplisafe3#354] [Q&A] [answered]
I am seemingly unable to pin the SimpliSafe plugin to a room and name in HomeKit. This bites me anytime my Homebridge service goes down, like a power outage. The device in HomeKit resets to SimpliSafe 3 and moves to another room. Any ideas on how to fix this?

**Comment by kdorepos** [comment]
I should add that I have created a room in Homebridge and have edited the name in Homebridge, as well.

**Comment by shamoon** [comment, marked as answer]
Do you have 'persistAccessories' checked (under advanced settings)? If so then the same accessory should persist between restarts etc, and in that case the name etc is handled entirely on the HomeKit side. If thats the case and still not working the issue has to be on the HK side (again, this plugin doesnt actually have anything to do with those settings e.g. room, name). So things like restarting the hub, signing in / out of iCloud or even re-creating your home (nuclear option) etc. can sometimes help with a 'stuck' accessory.

**Reply by kdorepos** [reply]
I think this was it. This seems to fix the issue. Thanks!
